roles/openshift_aws/tasks/vpc_and_subnet_id.yml

```diff
@@ -15,8 +15,14 @@
   ec2_vpc_subnet_facts:
     region: "{{ openshift_aws_region }}"
     filters:
-      "availability-zone": "{{ openshift_aws_subnet_az }}"
       vpc-id: "{{ vpcout.vpcs[0].id }}"
+      "tag:Name": "{{ item }}"
+  with_items: ['{%- if (openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list | length) == 1 -%}
```

Comment:
This is a tad unreadable, let's add a comment here to explain what we're doing.  Also, we could probably use a vars: block to build some of these tests instead of doing it in-line with jinja.

```yaml
with_items: "{{ l_subnet_list }}"
vars:
  l_multiple_list: "{{ openshift_aws_vpc.subnets[openshift_aws_region] | map(attribute="az") | list }}"
  l_use_multiple_subnets: "{{ (l_multiple_list | length == 1) | bool }}"
  l_subnet_list: "{{ l_use_multiple_subnets | ternary([openshift_aws_subnet_az], l_multiple_list) }}"
```

Reply:
We're removing all of this per conversation regarding removing default_az.